File: C++/array/prodArrayExceptSelf.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector <int> output;
        int currentProduct = 1;
        
        for (int i = 0; i < nums.size(); i++) {
            vector<int> temp = nums;
            temp.erase(temp.begin()+i);
            for (auto x: temp) {
                currentProduct = currentProduct * x;
            }
            output.push_back(currentProduct);
    
            currentProduct = 1;
        }
        
        return output;
        
    return {1};
     
    }
};

File: core/arrays/prod-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        product = 1
        output = [1] * len(nums)
        for i in range(len(nums)):
            output[i] *= product
            product *= nums[i]
            
        product = 1
        for i in reversed(range(len(nums))):
            output[i] *= product
            product *= nums[i]
            
        return output
